File: src/SkyBlock/enchants/sword/Backstabber.php
<?php


namespace SkyBlock\enchants\sword;


use pocketmine\event\entity\EntityDamageEvent;
use pocketmine\player\Player;

class Backstabber extends BaseMeleeEnchant {

    static int $id = 143;

    /**
     * @param Player $holder
     *
     * @return bool
     */
    public function isApplicableTo(Player $holder) : bool {
        return true;
    }

    /**
     * @param Player            $player
     * @param Player            $attacker
     * @param EntityDamageEvent $ev
     * @param int               $enchantmentlevel
     */
    public function onActivation(Player $player, Player $attacker, EntityDamageEvent $ev, int $enchantmentlevel) : void {
        if ($player->getDirectionVector()->dot($attacker->getDirectionVector()) > 0) {
            $this->sendActivation($player, "§bBackstabber §aActivated!");
            $this->sendActivation($attacker, "§cStruck by §bBackstabber §cEnchant!");
            $attacker->setHealth($attacker->getHealth() - (($enchantmentlevel - 1) / 100) * 4);

        }
    }
}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/sword/Backstabber.java
package dev.skyblock.enchants.sword;
import dev.skyblock.enchants.BaseMeleeEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
public class Backstabber extends BaseMeleeEnchant {
    public static final int ID = 143;
    public Backstabber() { super(ID, "Backstabber", 15); }
    @Override
    public boolean isApplicableTo(Player holder, int level) { return true; }
    @Override
    public void onActivation(Player victim, Player attacker, EntityDamageByEntityEvent ev, int level) {
        if (victim.getLocation().getDirection().dot(attacker.getLocation().getDirection()) > 0) {
            attacker.sendMessage("§bBackstabber §aActivated!");
            ev.setDamage(ev.getDamage() + ((level - 1) / 100.0) * 4);
        }
    }
}
